src/platform/renesas/SystemTimeSupport.cpp

```diff
@@ -0,0 +1,172 @@
+    {
+        // Note that sNumOverflows may be quite stale, and under those
+        // circumstances, the function may violate monotonicity guarantees
+        timeOut.xTimeOnEntering = xTaskGetTickCountFromISR();
+        timeOut.xOverflowCount  = sNumOfOverflows;
+    }
+    else
+    {
+#endif
+
+        vTaskSetTimeOutState(&timeOut);
+
+#ifdef __CORTEX_M
+        // BaseType_t is supposed to be atomic
+        sNumOfOverflows = timeOut.xOverflowCount;
+    }
+#endif
+
+    return static_cast<uint64_t>(timeOut.xTimeOnEntering) + (static_cast<uint64_t>(timeOut.xOverflowCount) << kTicksOverflowShift);
+}
+
+Clock::Microseconds64 ClockImpl::GetMonotonicMicroseconds64(void)
+{
+    return Clock::Microseconds64((FreeRTOSTicksSinceBoot() * kMicrosecondsPerSecond) / configTICK_RATE_HZ);
+}
+
+Clock::Milliseconds64 ClockImpl::GetMonotonicMilliseconds64(void)
+{
+    return Clock::Milliseconds64((FreeRTOSTicksSinceBoot() * kMillisecondsPerSecond) / configTICK_RATE_HZ);
+}
+
+uint64_t GetClock_Monotonic(void)
+{
+    return (FreeRTOSTicksSinceBoot() * kMicrosecondsPerSecond) / configTICK_RATE_HZ;
+}
+
+uint64_t GetClock_MonotonicMS(void)
+{
+    return (FreeRTOSTicksSinceBoot() * kMillisecondsPerSecond) / configTICK_RATE_HZ;
+}
+
+uint64_t GetClock_MonotonicHiRes(void)
+{
+    return GetClock_Monotonic();
+}
+
+CHIP_ERROR ClockImpl::GetClock_RealTime(Clock::Microseconds64 & aCurTime)
+{
+    // TODO(19081): This platform does not properly error out if wall clock has
+    //              not been set.  For now, short circuit this.
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+#if 0
+    if (sBootTimeUS == 0)
+    {
+        return CHIP_ERROR_REAL_TIME_NOT_SYNCED;
+    }
+    aCurTime = Clock::Microseconds64(sBootTimeUS + GetClock_Monotonic());
+    return CHIP_NO_ERROR;
+#endif
+}
+
+CHIP_ERROR ClockImpl::GetClock_RealTimeMS(Clock::Milliseconds64 & aCurTime)
+{
+    struct m_timeval tv;
+    clock_gettime(&tv);
+    aCurTime = std::chrono::seconds(tv.tv_sec) + std::chrono::milliseconds(tv.tv_usec/1000);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ClockImpl::SetClock_RealTime(Clock::Microseconds64 aNewCurTime)
+{
+    uint64_t timeSinceBootUS = GetClock_Monotonic();
+    if (aNewCurTime.count() > timeSinceBootUS)
+    {
+        sBootTimeUS = aNewCurTime.count() - timeSinceBootUS;
+    }
+    else
+    {
+        sBootTimeUS = 0;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR InitClock_RealTime()
+{
+    Clock::Microseconds64 curTime =
+        Clock::Microseconds64((static_cast<uint64_t>(CHIP_SYSTEM_CONFIG_VALID_REAL_TIME_THRESHOLD) * UINT64_C(1000000)));
+    // Use CHIP_SYSTEM_CONFIG_VALID_REAL_TIME_THRESHOLD as the initial value of RealTime.
+    // Then the RealTime obtained from GetClock_RealTime will be always valid.
+    //
+    // TODO(19081): This is broken because it causes the platform to report
+    //              that it does have wall clock time when it actually doesn't.
```

Comment:
Why are we enabling this when there are multiple issues with it ?

Reply:
This is solved by my reworked ipmlementation. Initializing and setting time is empty now as KaKu does sync time.